src/main/java/org/myrobotlab/framework/Service.java

```diff
@@ -2842,5 +2872,67 @@ public boolean isDev() {
     return false;
     
   }
-
+  
+  /**
+   * localize a key - details are
+   * http://myrobotlab.org/content/localization-myrobotlab-and-inmoov-languagepacks
+   * 
+   * @param key
+   * @return
+   */
+  public String localize(String key) {
+    return localize(key, (Object[])null);
+  }
+  
+  /**
+   * String format template processing localization
+   * 
+   * @param key
+   * @param args
+   * @return
+   */
+  public String localize(String key, Object ... args) {
+     if (key == null) {
+       log.error("localize(null) not allowed");
+       return null;
+     }     
+     key = key.toUpperCase();     
+     Object prop = localization.get(key);
+     
+     if (prop == null) {
+       prop = defaultLocalization.get(key);

```

Comment:
Will defaultLocalization ever be null?  I think you need to check that.. no?

Reply:
no .. it will not. The Service constructor loads localization and default if null will always be initialized to a map - even if en.properties is not found